### PR TITLE
node: fix --with-openssl by upgrading to openssl@1.1

### PR DESCRIPTION
**Pull request by chrmoritz**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Refs https://github.com/Homebrew/homebrew-core/pull/27036#issuecomment-384940826 and following comments.

I couldn't fully test the `emscripten` build, because I've not enough battery life left for the full source build. Let's hope it will work or I will fix the errors when I'm back home.

### CODE REVIEW

File: Formula/emscripten.rb [hunk 1]
```diff
@@ -46,12 +46,12 @@ class Emscripten < Formula
...
     ENV.cxx11
-    # OSX doesn't provide a "python2" binary so use "python" instead.
-    python2_shebangs = `grep --recursive --files-with-matches ^#!/usr/bin/.*python2$ #{buildpath}`
+    # rewrite hardcoded paths from system python to homebrew python
+    python2_shebangs = `grep --recursive --files-with-matches ^#!/usr/bin/python #{buildpath}`
     python2_shebang_files = python2_shebangs.lines.sort.uniq
     python2_shebang_files.map! { |f| Pathname(f.chomp) }
     python2_shebang_files.reject! &:symlink?
-    inreplace python2_shebang_files, %r{^(#!/usr/bin/.*python)2$}, "\\1"
+    inreplace python2_shebang_files, %r{^#!/usr/bin/python2?$}, "#!/usr/local/bin/python2"
```
Comment (ilovezfs) [inline Formula/emscripten.rb]: the prefix shouldn't be hard coded. also emscripten change should be in a separate pr since it needs new bottles whereas node doesn't

**Comment by ilovezfs** [comment]
Thanks @chrmoritz! Even though this will make `brew install emscripten` impossible for anyone who uses it, I guess it's still worth having the option for now.

**Comment by chrmoritz** [comment]
@ilovezfs Why that? I've installed the `emscripten` from #27212 without a build issue (and homebrew test passing) with a `node --with-openssl@1.1`.

**Comment by ilovezfs** [comment]
because it depends on openssl (via python@2) and openssl@1.1, and install is blocked if you have two different versions of a library in the dependency tree

**Comment by chrmoritz** [comment]
strange, it doesn't block for me

```
$ cat /usr/local/Cellar/node/10.0.0/INSTALL_RECEIPT.json                                                                                                              
{"homebrew_version":"1.6.2-33-g27ec9da","used_options":["--with-openssl@1.1"],"unused_options":["--with-debug","--without-npm","--without-completion","--without-icu4c"],"built_as_bottle":false,"poured_from_bottle":false,"installed_as_dependency":false,"installed_on_request":true,"changed_files":null,"time":1524828652,"source_modified_time":1524580904,"HEAD":"27ec9dae5fa4c2c5b19f8bfafaefbdbb3e9a9c8c","stdlib":null,"compiler":"clang","aliases":["nodejs","node@10","node.js","npm"],"runtime_dependencies":[{"full_name":"icu4c","version":"61.1"},{"full_name":"openssl@1.1","version":"1.1.0h"}],"source":{"path":"/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/node.rb","tap":"homebrew/core","spec":"stable","versions":{"stable":"10.0.0","devel":"","head":"HEAD","version_scheme":0}}}% 
$ brew remove emscripten                                                                                                                                                  
Uninstalling /usr/local/Cellar/emscripten/1.37.38... (9,381 files, 597.6MB)
$ brew install emscripten                                                                                                                                                 
==> Downloading https://homebrew.bintray.com/bottles/emscripten-1.37.38.high_sierra.bottle.tar.gz
==> Downloading from https://d29vzk4ow07wi7.cloudfront.net/a44c87271c85bbfb3ef49138f1f561454f6f34a842805433dffe6e86997f13c9?response-content-disposition=attachment%3Bfilename%3D%22emscripten-1.37.38.high_sierra.bottle.tar.gz%22&Policy=eyJTdGF0ZW1lbnQiOiBbeyJSZXNvdXJjZSI6Imh0dHA
######################################################################## 100.0%
==> Pouring emscripten-1.37.38.high_sierra.bottle.tar.gz
==> Caveats
Manually set LLVM_ROOT to
  /usr/local/opt/emscripten/libexec/llvm/bin
and comment out BINARYEN_ROOT
in ~/.emscripten after running `emcc` for the first time.
==> Summary
🍺  /usr/local/Cellar/emscripten/1.37.38: 9,381 files, 599.6MB
```

**Comment by ilovezfs** [comment]
@chrmoritz I've traced the issue to https://github.com/Homebrew/brew/pull/3654.

**Comment by ilovezfs** [comment]
@chrmoritz I've opened https://github.com/Homebrew/brew/issues/4124

**Comment by lopsch** [comment]
Hi guys,

I think this is related, right:

```
node (10.0.0) < 10.1.0
==> Upgrading 1 outdated package, with result:
node 10.0.0 -> 10.1.0
Error: node contains conflicting version recursive dependencies:
  openssl@1.1, openssl
View these with `brew deps --tree node`.
```

I have python@2 installed which depends on openssl and node depends on openssl@1.1 and python@2. Is there a way out of this :-)?

Thanks and best regards
